front/src/components/ui/button.tsx

```diff
@@ -40,7 +40,7 @@ export interface ButtonProps
 const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
 	({ className, variant, size, asChild = false, ...props }, ref) => {
 		const Comp = asChild ? Slot : "button";
-		return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
+		return <Comp className={cn(buttonVariants({ variant, size }), className)} ref={ref} {...props} />;
```

Comment:
won't this pose problems if we want to change button variants with classes ?

Reply:
i don't think so, by inserting className into buttonVariants the button defaults are used as primary by cva() and are not overriden by tailwind merge, whereas they are if they are passed as second parameter. I saw it done like this on other components I implemented in this PR and changed it here